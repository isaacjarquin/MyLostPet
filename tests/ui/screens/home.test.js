import "react-native"
import React from "react"
import Home from "../../../screens/home"
import { shallow } from "enzyme"
import toJSON from "enzyme-to-json"

describe("Home", () => {
    const navigate = jest.fn()

    const props = {
        navigation: {
            navigate: navigate
        }
    }

    beforeEach(() => {
        wrapper = shallow(<Home {...props} />)
    })

    afterEach(() => {
        navigate.mockReset()
    })

    it("renders correctly", () => {
        expect(wrapper).toBeDefined()
        expect(toJSON(wrapper)).toMatchSnapshot()
    })

    describe("navigateToMissingPetForm", () => {
        beforeEach(() => {
            wrapper.instance().navigateToMissingPetForm()
        })

        it("navigate to MissingPetForm", () => {
            expect(navigate).toHaveBeenCalledTimes(1)
            expect(navigate).toHaveBeenCalledWith("MissingPetForm")
        })
    })

    describe("navigateToSearchForm", () => {
        beforeEach(() => {
            wrapper.instance().navigateToSearchForm()
        })

        it("navigate to MissingPetForm", () => {
            expect(navigate).toHaveBeenCalledTimes(1)
            expect(navigate).toHaveBeenCalledWith("SearchForm")
        })
    })

    describe("_showModal", () => {
        it("modify isModalVisible state when called", () => {
            wrapper.setState({ isModalVisible: false })
            wrapper.instance()._showModal()
            expect(wrapper.state().isModalVisible).toBe(true)
        })
    })
})