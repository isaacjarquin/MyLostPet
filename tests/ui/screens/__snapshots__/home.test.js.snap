// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#fff",
      "flex": 1,
      "flexDirection": "column",
      "height": 1334,
      "justifyContent": "space-between",
      "width": 750,
    }
  }
>
  <ImageBackground
    source={
      Object {
        "testUri": "../../../assets/images/home.jpg",
      }
    }
    style={
      Object {
        "flex": 1,
        "height": 1334,
        "justifyContent": "flex-end",
        "width": 750,
      }
    }
  >
    <View
      style={
        Object {
          "flex": 1,
          "marginTop": 30,
          "minHeight": 944,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "alignSelf": "center",
            "backgroundColor": "transparent",
            "color": "grey",
            "fontSize": 18,
            "marginBottom": 20,
          }
        }
      >
        Te ayudamos a encontrarlo
      </Text>
      <Image
        source={
          Object {
            "testUri": "../../../assets/images/pet-care-icon.png",
          }
        }
        style={
          Object {
            "alignSelf": "center",
            "height": 100,
            "width": 100,
          }
        }
      />
    </View>
    <View
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <TouchableOpacity
        activeOpacity={0.2}
        onPress={[Function]}
        style={
          Object {
            "backgroundColor": "#333333",
            "borderRadius": 3,
            "flexDirection": "row",
            "marginLeft": 5,
            "marginRight": 5,
            "marginTop": 5,
            "opacity": 0.8,
            "padding": 10,
          }
        }
      >
        <Icon
          color="white"
          name="plus"
          raised={false}
          reverse={false}
          reverseColor="white"
          size={30}
          style={
            Object {
              "alignSelf": "center",
              "marginLeft": "15%",
            }
          }
          type="entypo"
          underlayColor="white"
        />
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "alignSelf": "center",
              "backgroundColor": "transparent",
              "color": "white",
              "fontSize": 18,
              "margin": 10,
            }
          }
        >
          AÃ±adir nueva mascota
        </Text>
      </TouchableOpacity>
      <SecondaryMenuModal
        navigate={[Function]}
      />
    </View>
    <SafeAreaView
      style={
        Object {
          "marginTop": 10,
        }
      }
    />
  </ImageBackground>
</View>
`;
